pkg/openshift/webconsole.go

```diff
@@ -0,0 +1,61 @@
+package openshift
+
+//go:generate go run ./.packr/packr.go
+
+import (
+	"context"
+	"github.com/RHsyseng/operator-utils/pkg/logs"
+	"github.com/RHsyseng/operator-utils/pkg/utils/kubernetes"
+	"github.com/RHsyseng/operator-utils/pkg/utils/openshift"
+	"github.com/ghodss/yaml"
+	"github.com/gobuffalo/packr/v2"
+	hawtiov1alpha1 "github.com/hawtio/hawtio-operator/pkg/apis/hawtio/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logs.GetLogger("openshift-webconsole")
+
+func ConsoleYAMLSampleExists() error {
+	gvk := schema.GroupVersionKind{Group: "console.openshift.io", Version: "v1", Kind: "ConsoleYAMLSample"}
+	return kubernetes.CustomResourceDefinitionExists(gvk)
+}
+
+func CreateConsoleYAMLSamples(c client.Client) {
+	log.Info("Loading CR YAML samples.")
+	box := packr.New("cryamlsamples", "../../deploy/crs")
+	if box.List() == nil {
+		log.Error("CR YAML folder is empty. It is not loaded.")
+		return
+	}
+
+	resMap := make(map[string]string)
+	for _, filename := range box.List() {
+		yamlStr, err := box.FindString(filename)
+		if err != nil {
+			resMap[filename] = err.Error()
+			continue
+		}
+		hawtio := hawtiov1alpha1.Hawtio{}
+		err = yaml.Unmarshal([]byte(yamlStr), &hawtio)
+		if err != nil {
+			resMap[filename] = err.Error()
+			continue
+		}
+		yamlSample, err := openshift.GetConsoleYAMLSample(&hawtio)
```

Comment:
Would it be possible to add a variable that could be used at build time to customise the name of the sample? That would be used in the product build pipeline to change the name.

Reply:
Currently it is possible to customize the sample name, title, desc. snippet or not by adding annotations in the yaml sample file.  You can check out the sample I checked in.  Hope that is something you were expecting.

Reply:
Ah, I didn't notice these annotations. Are they actually used somewhere, or are they only informational?

My concern is for the product operator, for which some defaults are changed to reflect the product name, rather than that of the upstream project. We already have a couple of Go build-time variables that are used to address that concern:

https://github.com/hawtio/hawtio-operator/blob/4183906c78c01df6e981de33682c3493cfcb51c1/pkg/controller/hawtio/hawtio_controller.go#L72-L74

My idea was to add one, so for the sample name. WDYT?

Reply:
Those annotations in the sample are used to customize the YAML sample during the creation time by this new func in webconsole.go.  They will be taken out from the YAML created in OCP.

Do you mean the product name: "hawt.io" in each yaml file may be changed for each build?
And the new variable you mentioned is to replace the ones in the all the yaml files?
Do you prefer to have it as environment like that ImageRepository?

Reply:
> Those annotations in the YAML sample are used to customize the YAML sample during the creation time. They will be taken out from the YAML sample created in OCP.

Thanks for the clarification. May I suggest to choose a naming scheme that is more qualifying for these annotations? It's not clear at the moment for a reader what is the purpose of these annotations. That makes me wonder what is the value compared to simply packaging the real sample, that is using the `ConsoleYAMLSample` API directly?

> Do you mean the product name: "hawt.io" may be changed for each build?
And the new variable you mentioned has the value to replace all the occurrences in those yaml sample files?
Do you prefer to have it as environment like that ImageRepository?

I mean the `metadata.name` of the `Hawtio` resource inside the sample. Ideally, it should be using the product name when it's created with the productised version of the operator. That is what we do currently in the sample from the CSV.